framework/decode/dx12_pre_process_consumer.h

```diff
@@ -20,17 +20,20 @@
 ** DEALINGS IN THE SOFTWARE.
 */
 
-#ifndef GFXRECON_DECODE_DX12_BROWSE_CONSUMER_H
-#define GFXRECON_DECODE_DX12_BROWSE_CONSUMER_H
+#ifndef GFXRECON_DECODE_DX12_PRE_PROCESS_CONSUMER_H
+#define GFXRECON_DECODE_DX12_PRE_PROCESS_CONSUMER_H
 
-#include <memory>
-
-#include "generated/generated_dx12_decoder.h"
-#include "decode/dx12_dump_resources.h"
+#include "generated/generated_dx12_consumer.h"
 
 GFXRECON_BEGIN_NAMESPACE(gfxrecon)
 GFXRECON_BEGIN_NAMESPACE(decode)
 
+#define CHECK_DX12_CONSUMER_USAGE() \
+    if (!dx12_consumer_usage_)      \
+    {                               \
+        return;                     \
+    }
+
```

Comment:
This macro is can only be used inside the Dx12PreProcessConsumer class. Maybe it should be defined inside that class? I realized that it being defined inside that class doesn't prevent it from being invoked outside the class, but it does make it clear that is where it is intended to be used.)

Reply:
Good idea. Moved.